src/glue/jobs/s3_to_json.py

```diff
@@ -16,6 +16,237 @@
+
+    For JSON whose data type is "EnrolledParticipants":
+        - Cast "CustomFields.Symptoms" and "CustomFields.Treatments" property
+          to an array. (May be an empty array).
+
+    For relevant Garmin data types:
+        - Some Garmin data types have one or more properties which are objects
+          (usually mapping time to some metric), that would be better formatted
+          as an array of objects. We transform these properties into arrays of
+          objects.
+
+    Args:
+        json_obj (str): A JSON object sourced from the JSON file of this data type.
+        dataset_identifier (str): The data type of `json_obj`.
+        metadata (dict): Metadata derived from the file basename.
+
+    Returns:
+        json_obj (dict) The JSON object with the relevant transformations applied.
+    """
+    logger = logging.getLogger(__name__)
+    if metadata["start_date"] is not None:
+        json_obj["export_start_date"] = metadata["start_date"].isoformat()
+    else:
+        json_obj["export_start_date"] = None
+    json_obj["export_end_date"] = metadata.get("end_date").isoformat()
+    if dataset_identifier in DATA_TYPES_WITH_SUBTYPE:
+        # This puts the `Type` property back where Apple intended it to be
+        json_obj["Type"] = metadata["subtype"]
+    if dataset_identifier == "SymptomLog":
+        # Load JSON string as dict
+        json_obj["Value"] = json.loads(json_obj["Value"])
+    if dataset_identifier == "EnrolledParticipants":
+        for field_name in ["Symptoms", "Treatments"]:
+            if (
+                    field_name in json_obj["CustomFields"]
+                    and isinstance(json_obj["CustomFields"][field_name], str)
+               ):
+                if len(json_obj["CustomFields"][field_name]) > 0:
+                    # This JSON string was written in a couple different ways
+                    # in the testing data: "[{\\\"id\\\": ..." and "[{\"id\": ..."
+                    # or just an empty string. It's not really clear which format
+                    # is intended, (The example they provided has it written as
+                    # an object rather than a string, so...).
+                    try:
+                        json_obj["CustomFields"][field_name] = json.loads(
+                                json_obj["CustomFields"][field_name]
+                        )
+                    except json.JSONDecodeError as e:
+                        # If it's not propertly formatted JSON, then we
+                        # can't read it, and instead store an empty list
+                        logger.warning(f"Problem CustomFields.{field_name}: "
+                                       f"{json_obj['CustomFields'][field_name]}")
+                        logger.warning(str(e))
+                        json_obj["CustomFields"][field_name] = []
+                else:
+                    json_obj["CustomFields"][field_name] = []
+    garmin_transform_types = {
```

Comment:
I like what you've done here, but is it higher risk in maintenance down the line if we forget that the first tuple has to be the key and the second tuple is value?